### PR TITLE
feat: Remove reinterpret_casts from AG race timer script and add method and chat command to get current server uptime

### PR DESCRIPTION
**Pull request by jadebenn**
Tested in game by running the race and verified that mission 1884 is correctly marked as completed when race is finished

### CODE REVIEW

**Review by EmosewaMC — DISMISSED** [review]
and the achievements only work if you get below the target time as well?

**Comment by jadebenn** [comment]
Marking this as draft as I want to modify the implementation and run a few more tests

**Comment by jadebenn** [comment]
Tested that achievements work correctly

File: dScripts/02_server/Map/AG/NpcAgCourseStarter.cpp [hunk 1]
```diff
@@ -15,83 +15,94 @@ void NpcAgCourseStarter::OnUse(Entity* self, Entity* user) {
 	auto* const scriptedActivityComponent = self->GetComponent<ScriptedActivityComponent>();
 	if (!scriptedActivityComponent) return;
 
-	if (scriptedActivityComponent->GetActivityPlayerData(user->GetObjectID()) != nullptr) {
-		GameMessages::SendNotifyClientObject(self->GetObjectID(), u"exit", 0, 0, LWOOBJID_EMPTY, "", user->GetSystemAddress());
+	const auto selfId = self->GetObjectID();
+	const auto userId = user->GetObjectID();
+	const auto userSysAddr = user->GetSystemAddress();
```
Comment (EmosewaMC) [inline dScripts/02_server/Map/AG/NpcAgCourseStarter.cpp:20]:
```suggestion
	const auto& userSysAddr = user->GetSystemAddress();
```

File: dScripts/02_server/Map/AG/NpcAgCourseStarter.cpp [hunk 2]
```diff
@@ -15,83 +15,94 @@ void NpcAgCourseStarter::OnUse(Entity* self, Entity* user) {
...
 	}
 }
 
 void NpcAgCourseStarter::OnMessageBoxResponse(Entity* self, Entity* sender, int32_t button, const std::u16string& identifier, const std::u16string& userData) {
 	auto* const scriptedActivityComponent = self->GetComponent<ScriptedActivityComponent>();
 	if (!scriptedActivityComponent) return;
 
+	const auto selfId = self->GetObjectID();
+	const auto senderId = sender->GetObjectID();
+	const auto senderSysAddr = sender->GetSystemAddress();
```
Comment (EmosewaMC) [inline dScripts/02_server/Map/AG/NpcAgCourseStarter.cpp:35]:
```suggestion
	const auto& senderSysAddr = sender->GetSystemAddress();
```

File: dScripts/02_server/Map/AG/NpcAgCourseStarter.cpp [hunk 3]
```diff
@@ -15,83 +15,94 @@ void NpcAgCourseStarter::OnUse(Entity* self, Entity* user) {
...
 }
 
 void NpcAgCourseStarter::OnFireEventServerSide(Entity* self, Entity* sender, std::string args, int32_t param1, int32_t param2, int32_t param3) {
 	auto* const scriptedActivityComponent = self->GetComponent<ScriptedActivityComponent>();
 	if (!scriptedActivityComponent) return;
 
-	auto* const data = scriptedActivityComponent->GetActivityPlayerData(sender->GetObjectID());
+	const auto selfId = self->GetObjectID();
+	const auto senderId = sender->GetObjectID();
+	const auto senderSysAddr = sender->GetSystemAddress();
```
Comment (EmosewaMC) [inline dScripts/02_server/Map/AG/NpcAgCourseStarter.cpp:75]:
```suggestion
	const auto& senderSysAddr = sender->GetSystemAddress();
```

File: docs/Commands.md [hunk 4]
```diff
@@ -45,6 +45,7 @@
 |setannmsg|`/setannmsg <title>`|Sets the message of an announcement.|8|
 |setanntitle|`/setanntitle <title>`|Sets the title of an announcement.|8|
 |shutdownuniverse|`/shutdownuniverse`|Sends a shutdown message to the master server. This will send an announcement to all players that the universe will shut down in 10 minutes.|9|
+|uptime|`/uptime`|Displays the time the current world server has been active.|10|
```
Comment (EmosewaMC) [inline docs/Commands.md:48]:
```suggestion
|uptime|`/uptime`|Displays the time the current world server has been active.|1|
```
maybe move it up to the other gm 1 commands too?

**Review by EmosewaMC — DISMISSED** [review]

File: docs/Commands.md (hunk 4 above)
Comment (jadebenn) [reply docs/Commands.md]: I set the level to 8, but kept it where it is

File: dScripts/02_server/Map/AG/NpcAgCourseStarter.cpp (hunk 1 above)
Comment (jadebenn) [reply dScripts/02_server/Map/AG/NpcAgCourseStarter.cpp]: 👍

File: dScripts/02_server/Map/AG/NpcAgCourseStarter.cpp (hunk 2 above)
Comment (jadebenn) [reply dScripts/02_server/Map/AG/NpcAgCourseStarter.cpp]: 👍

File: dScripts/02_server/Map/AG/NpcAgCourseStarter.cpp (hunk 3 above)
Comment (jadebenn) [reply dScripts/02_server/Map/AG/NpcAgCourseStarter.cpp]: 👍

**Review by EmosewaMC — APPROVED** [review]

**Review by EmosewaMC — APPROVED** [review]

**Comment by EmosewaMC** [comment]
i forgot if i approved this or not

**Review by EmosewaMC — APPROVED** [review]
just in case

**Review by aronwk-aaron — APPROVED** [review]

**Pull request merged**